### PR TITLE
feat: support for aws lambda batch feature

### PR DESCRIPTION
**Pull request by ShreyaGangishetty**
*Issue #, if available:*

*Description of changes:*
Added support for a new property (MaximumBatchingWindowInSeconds) of AWS Lambda EventSourceMapping

*Description of how you validated changes:*

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [x] Update documentation
- [x] Verify transformed template deploys and application functions as expected
- [x] Add/update example to `examples/2016-10-31`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/translator/input/kinesis_stream.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            }
+        }
+      Runtime: nodejs8.10
+      Timeout: 10
+      Events:
+        Stream:
+          Type: Kinesis
+          Properties:
+            Stream: !GetAtt GreetingStream.Arn
+            MaximumBatchingWindowInSeconds: 20
```
Comment (keetonian) [inline tests/translator/input/kinesis_stream.yaml]: Can you test a `!Ref Parameter` here as well to make sure it can handle parameter references?

Comment (ShreyaGangishetty) [reply tests/translator/input/kinesis_stream.yaml]: Added the parameter reference and it is working as expected.
```yaml
Parameters:
  MyBatchingWindowParam:
    Type: Number
    Default: 45
    Description: parameter for batching window in seconds
Resources:
  ...
      Events:
        Stream:
            ...
            MaximumBatchingWindowInSeconds: !Ref MyBatchingWindowParam
...
```

File: tests/translator/test_translator.py [hunk 2]
```diff
@@ -259,6 +259,7 @@ class TestTranslatorEndToEnd(TestCase):
         'api_with_apikey_default_override',
         'api_with_apikey_required',
         'api_with_path_parameters',
+        'kinesis_stream'
```
Comment (praneetap) [inline tests/translator/test_translator.py]: Please follow the naming convention "function_with_batch_window...."

Comment (ShreyaGangishetty) [reply tests/translator/test_translator.py]: renamed 'kinesis_stream' with 'function_with_batch_window'

**Comment by ShreyaGangishetty** [comment]
> Could you also update the docs for these event types?
> 
> Kinesis: https://github.com/awslabs/serverless-application-model/blob/develop/versions/2016-10-31.md#kinesis
> DynamoDB: https://github.com/awslabs/serverless-application-model/blob/develop/versions/2016-10-31.md#dynamodb
> SQS: https://github.com/awslabs/serverless-application-model/blob/develop/versions/2016-10-31.md#sqs

Updated the docs

File: tests/translator/input/kinesis_stream.yaml (hunk 1 above)
Comment (keetonian) [reply tests/translator/input/kinesis_stream.yaml]: I don't see this in the tests

**Pull request merged**